src/extension.ts

```diff
@@ -70,6 +70,9 @@ export function activate(context: vscode.ExtensionContext) {
         }
       };
 
+      //Add Plugin-Version and Security-Vendor as custom header fields
+      process.env['CUSTOM_HEADER'] = '{"plugin_version":"' + version + '", "security_vendor":"' + vendor + '"}';

```

Comment:
You can use multi line string
```
process.env['CUSTOM_HEADER'] = `{"plugin_version: "${version}", "security_vendor":"${vendor}"}`;
```
Also HTTP headers uses hyphen(-) to separate words,

e.g. Content-Type

In this case you may have to rename to `Plugin-Version`.

I'm fine with underscore(_) separation If it is already decided and coded according in the backend. If you decide to go with underscore(_), then I would suggest to put the custom header inside constants.ts as normal javascript object and use `JSON.stringify` while passing it through env variable.